geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HdelDUnitTest.java

```diff
@@ -0,0 +1,165 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+import io.lettuce.core.ClientOptions;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.api.StatefulRedisConnection;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.resource.ClientResources;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import org.apache.geode.internal.AvailablePortHelper;
+import org.apache.geode.redis.ConcurrentLoopingThreads;
+import org.apache.geode.redis.session.springRedisTestApplication.config.DUnitSocketAddressResolver;
+import org.apache.geode.test.dunit.rules.MemberVM;
+import org.apache.geode.test.dunit.rules.RedisClusterStartupRule;
+import org.apache.geode.test.junit.rules.ExecutorServiceRule;
+
+public class HdelDUnitTest {
+
+  @ClassRule
+  public static RedisClusterStartupRule cluster = new RedisClusterStartupRule();
+
+  @ClassRule
+  public static ExecutorServiceRule executor = new ExecutorServiceRule();
+
+  private static final int HASH_SIZE = 50000;
+  private static MemberVM locator;
+  private static MemberVM server1;
+  private static MemberVM server2;
+  private static int[] redisPorts;
+  private static RedisCommands<String, String> lettuce;
+  private static StatefulRedisConnection<String, String> connection;
+  private static ClientResources resources;
+
+  @BeforeClass
+  public static void classSetup() {
+    redisPorts = AvailablePortHelper.getRandomAvailableTCPPorts(3);
+
+    String redisPort1 = "" + redisPorts[0];
+    String redisPort2 = "" + redisPorts[1];
+
+    locator = cluster.startLocatorVM(0);
+
+    server1 = startRedisVM(1, redisPorts[0]);
+    server2 = startRedisVM(2, redisPorts[1]);
+
+    DUnitSocketAddressResolver dnsResolver =
+        new DUnitSocketAddressResolver(new String[] {redisPort2, redisPort1});
+
+    resources = ClientResources.builder()
+        .socketAddressResolver(dnsResolver)
+        .build();
+
+    RedisClient redisClient = RedisClient.create(resources, "redis://localhost");
+    redisClient.setOptions(ClientOptions.builder()
+        .autoReconnect(true)
+        .build());
+    connection = redisClient.connect();
+    lettuce = connection.sync();
+  }
+
+  private static MemberVM startRedisVM(int vmID, int redisPort) {
+    int locatorPort = locator.getPort();
+
+    return cluster.startRedisVM(vmID, x -> x
+        .withConnectionToLocator(locatorPort)
+        .withProperty(REDIS_PORT, "" + redisPort));
+  }
+
+  @Before
+  public void testSetup() {
+    lettuce.flushall();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    resources.shutdown().get();
+    connection.close();
+    server1.stop();
+    server2.stop();
+  }
+
+  @Test
+  public void testConcurrentHDelReturnExceptedNumberOfDeletions() {
```

Comment:
Since you'll have to re-trigger anyway, the name could be changed to correct some grammatical errors/fit in with the test name below it: `testConcurrentHDel_returnsExpectedNumberOfDeletions`

Reply:
Thanks! Done.